pages/product/[name].js

```diff
@@ -0,0 +1,104 @@
+import Link from 'next/link'
+import { useState, useEffect } from 'react'
+import { StarIcon } from '@heroicons/react/solid'
+import { HeartIcon, StarIcon as StarIconOutline } from '@heroicons/react/outline'
+
+const Product = ({ data }) => {
+  const [allProducts, setAllProducts] = useState([])
+  const [selectedImage, setSelectedImage] = useState(`https://opt.moovweb.net/?quality=50&img=${data.images[0].url}`)
```

Comment:
why not just store the index and build the url down in the component? makes your setSelectedImage calls much cleaner.